src/mods/inputdisp.cpp

```diff
@@ -170,26 +182,52 @@ static bool get_notch_pos(const MergedStickInputs& stickInputs, Vec2d* out_pos)
 }
 
 static const mkb::GXColor s_color_map[] = {
-    {0xb1, 0x5a, 0xff, 0xff},  // Purple
+    draw::PURPLE,              // Purple
     draw::RED,                 // Red
     draw::ORANGE,              // Orange
     {0xfd, 0xfb, 0x78, 0xff},  // Yellow
     {0x78, 0xfd, 0x85, 0xff},  // Green
     {0x78, 0xca, 0xfd, 0xff},  // Blue
     draw::PINK,                // Pink
-    {0x00, 0x00, 0x00, 0xff},  // Black
+    draw::BLACK,               // Black
 };
 
-static void draw_stick(const MergedStickInputs& stickInputs, const Vec2d& center, f32 scale) {
-    mkb::GXColor chosen_color = s_color_map[pref::get(pref::U8Pref::InputDispColor)];
+static mkb::GXColor get_color() {
+    InputDispColorType color_pref = InputDispColorType(pref::get(pref::U8Pref::InputDispColorType));
+    switch (color_pref) {
+        case InputDispColorType::Default: {
+            return s_color_map[pref::get(pref::U8Pref::InputDispColor)];
+        }
+        case InputDispColorType::RGB: {
+            return {
+                .r = pref::get(pref::U8Pref::InputDispRed),
+                .g = pref::get(pref::U8Pref::InputDispGreen),
+                .b = pref::get(pref::U8Pref::InputDispBlue),
+                .a = 0xff,
+            };
+        }
+        case InputDispColorType::Rainbow: {
+            return draw::num_to_rainbow(s_rainbow);
+        }
+        case InputDispColorType::MatchBall: {
+            mkb::GXColor current = ballcolor::get_current_color();
+            current.a = 0xff;
+            return current;
+        }
+    }
+    return draw::BLACK;  // shouldnt reach
```

Comment:
Could make this an assertion (but I probably don't do this consistently either)

Reply:
Added the assertion